### DISCUSSION TITLE
AIO 8.0.0 - override config.php with "::1" at trusted_proxies

### DISCUSSION THREAD

**Discussion by Rimos911** [nextcloud/all-in-one#4432] [Questions] [answered]
I have noticed that since the Nextcloud AIO v8.0.0 update, after the nightly maintenance or restart the container, NC permanently override or add "1 => '::1'," to the array 'trusted_proxies'. As a result, I get the yellow-message (self Check) that my reverse proxy is not configured correctly. Until now I have always had a "green" check status.

My Nextcloud AIO runs on his own Server (Ubuntu + Docker + Portainer) - the reverse proxy runs on a Synology Diskstation.

before (config.php)

'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '192.168.2.18',

after restart (config.php)

  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '::1',
  ),

As a result, I get the yellow-message that, my reverse proxy is not configured correctly.

Could it be a bug that NC overide this configuration. I use NC over 2 Years an I dont had this problem before. Thank you!

**Comment by szaimen** [comment, marked as answer]
Hi, if you add your ip-address like so, it should be kept:

`2 => '192.168.2.18',`

**Reply by Rimos911** [reply]
No, Not only - i have to delete the line with "::1" than it works. But After Night it is the Same. The cycle adds ::1 and i have the same Problem.

I can Not deaktivate to add the ipv6 localhost. I don't use ipv6 in my Network. Ur is disabled.

**Reply by szaimen** [reply]
I think this should still work even if ipv6 is disabled. Since php is compiled with ipv6 support or did you disable ipv6 on your host in the kernel?

Can you open an issue here: https://github.com/nextcloud/server/issues/new/choose

**Reply by lastsamurai26** [reply]
for me i disabled it directly on the host himself

**Reply by Rimos911** [reply]
How exactly did you do that? can you post instructions. I want to try to replicate it.

Thank you!

**Reply by lastsamurai26** [reply]
i added this to my /etc/sysctl.conf
```
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
net.ipv6.conf.tun0.disable_ipv6 = 1
```

Inside of the Nextcloud-aio-nextcloud container 
inet addr:172.22.0.8  Bcast:172.22.255.255  Mask:255.255.0.0
inet6 addr: fe80::42:acff:fe16:8/64 Scope:Link

```
docker inspect network nextcloud-aio
[
    {
        "Name": "nextcloud-aio",
        "Id": "523697c983fbaae7c1dbfa6ca18907830dca06d9dd2c9ea822d8243c60226c5c",
        "Created": "2024-02-27T15:55:43.748378209+01:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,

```
the Docker network don't display a IPv6 
```
            "f02b950f3322cd9d3510a37b2cb6cc37d80b4b65f4f913fe14fc4621984f6e26": {
                "Name": "nextcloud-aio-nextcloud",
                "EndpointID": "0b6e1289e78041bad59338d7b8dd601758c97ed2abd92ef7ac0b4c5d6575b1c3",
                "MacAddress": "02:42:ac:16:00:08",
                "IPv4Address": "172.22.0.8/16",
                "IPv6Address": ""
            },

```

```
Client: Docker Engine - Community
 Version:           26.0.0
 API version:       1.45
 Go version:        go1.21.8
 Git commit:        2ae903e
 Built:             Wed Mar 20 15:18:01 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          26.0.0
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       8b79278
  Built:            Wed Mar 20 15:18:01 2024
  OS/Arch:          linux/amd64
  Experimental:     false

 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Reply by Rimos911** [reply]
Thanks for the fast reply.

I´ve try it, IPv6 is disabled but after restart all containers NC add the line "1 => ´::1', again and erase the line with the local IP

before:
  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    2 => '192.168.2.18',

after restart:
  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    2 => '192.168.2.18',
    1 => '::1',
  ),

Again "::1" was added an NC has (on my System) a problem with the local-IP-V6 - yellow check.

Is it the [nextcloud-aio-mastercontainer] that modify the config.php? What is the reason?

**Reply by lastsamurai26** [reply]
can you check if your nextcloud-aio-nextcloud container have a ipv6 ? 
normally it should not but for some reason my have one

**Reply by Rimos911** [reply]
IPv6 is inside The Container also disabled.

**Reply by lastsamurai26** [reply]
strange i don't know why it is enabled in my container .

**Comment by Rimos911** [comment]
Hi,

I did that. Nextcloud has a problem with the "::1"

my current configuration (config.php):

  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '::1',
    2 => '192.168.2.18',
  ),

This is the configuration of the Synology DSM Reverse Proxy Settings:
![Screenshot 2024-03-25 101856](https://github.com/nextcloud/all-in-one/assets/70541883/8fddccfc-2373-464c-9c89-06b2c5c0727d)
![Screenshot 2024-03-25 101926](https://github.com/nextcloud/all-in-one/assets/70541883/4eb5c801-ec3f-4b54-8c4a-01c416c898a8)

As I said, it worked without any problems until about 2 weeks ago

If I erase the "::1" the Check was green, but after night it is added again.

**Reply by lastsamurai26** [reply]
I have the Same Problem. 

My Server doesn't have a IPv6 because  I have deactivated this in the system, but I have noticed that the Docker container itself has one although it should not have it. After the update to version 8, however, it now has one even though system-wide IPv6 is deactivated

**Comment by Rimos911** [comment]
I found the reason... IPv6 was enabled on the Server. I´ve disabled IPv6... everything works and selfcheck is green.

The only thing I noticed is that 3 containers are yellow - they are unhealthy now but NC seems to run without
visibel problems.

https://github.com/nextcloud/all-in-one/discussions/4508
